service/collector/usage.go

```diff
@@ -100,52 +98,33 @@ func NewUsage(config UsageConfig) (*Usage, error) {
 
 		usageScrapeError: scrapeErrorCounter,
 
-		environmentName: config.EnvironmentName,
-		location:        config.Location,
+		environmentName:        config.EnvironmentName,
+		location:               config.Location,
+		cpAzureClientSetConfig: config.CPAzureClientSetConfig,
 	}
 
 	return u, nil
 }
 
 func (u *Usage) Collect(ch chan<- prometheus.Metric) error {
-	// We need all CRs to gather all subscriptions below.
-	var crs []providerv1alpha1.AzureConfig
-	{
-		mark := ""
-		page := 0
-		for page == 0 || len(mark) > 0 {
-			opts := metav1.ListOptions{
-				Continue: mark,
-			}
-			list, err := u.g8sClient.ProviderV1alpha1().AzureConfigs("").List(opts)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-
-			crs = append(crs, list.Items...)
-
-			mark = list.Continue
-			page++
-		}
+	ctx := context.Background()
+	clientSets, err := credential.GetAzureClientSetsFromCredentialSecretsBySubscription(u.k8sClient, u.environmentName)
```

Comment:
This collector will still expose metrics only by subscription. Resource groups are the same even using different client id or tenant id.